scripts/cluster/agent.py

```diff
@@ -423,6 +443,285 @@ def configure():
+    return output
+
+
+@app.route('/{}/service/disable'.format(CLUSTER_API), methods=['POST'])
+def service_disable():
+    """
+    Web call to enable a service
+    :request_param service: the name of the service
+    """
+    v = callback_token_validation(request)
+    if not v["valid"]: return v["response"]
+    if "service" not in request.json or len(request.json["service"].strip()) == 0:
+        error_msg = {"error": "Empty service provided"}
+        return Response(json.dumps(error_msg), mimetype='application/json', status=400)
+    output = subprocess.check_output("systemctl disable snap.{}.service".format(request.json["service"]).split())
+    return output
+
+
+@app.route('/{}/service/logs'.format(CLUSTER_API), methods=['POST'])
+def service_logs():
+    """
+    Web call to the logs of a service
+    :request_param service: the name of the service
+    :request_param lines: total lines | if omitted default value 10 will be used
+    """
+    lines = 10
+    v = callback_token_validation(request)
+    if not v["valid"]: return v["response"]
+    if "service" not in request.json or len(request.json["service"].strip()) == 0:
+        error_msg = {"error": "Empty service provided"}
+        return Response(json.dumps(error_msg), mimetype='application/json', status=400)
+    if "lines" in request.json:
+        lines = request.json["lines"]
+    # more info on logs could be obtained by using: --output=json-pretty
+    output = subprocess.check_output("journalctl --lines={} --unit=snap.{}.service --no-pager".format(lines, request.json["service"]).split())
+    return output
+
+
+@app.route('/{}/addon/enable'.format(CLUSTER_API), methods=['POST'])
+def enable():
+    """
+    Web call to microk8s.enable <addon>
+    :request_param addon: the name of the addon
+    """
+    v = callback_token_validation(request)
+    if not v["valid"]: return v["response"]
+    if "addon" not in request.json or len(request.json["addon"].strip()) == 0:
+        error_msg = {"error": "Empty addon provided"}
+        return Response(json.dumps(error_msg), mimetype='application/json', status=400)
+    output = subprocess.check_output("{}/microk8s-enable.wrapper {}".format(snap_path, request.json["addon"]).split())
+    return output
+
+
+@app.route('/{}/addon/disable'.format(CLUSTER_API), methods=['POST'])
+def disable():
+    """
+    Web call to microk8s.disable <addon>
+    :request_param addon: the name of the addon
+    """
+    v = callback_token_validation(request)
+    if not v["valid"]: return v["response"]
+    if "addon" not in request.json or len(request.json["addon"].strip()) == 0:
+        error_msg = {"error": "Empty addon provided"}
+        return Response(json.dumps(error_msg), mimetype='application/json', status=400)
+    output = subprocess.check_output("{}/microk8s-disable.wrapper {}".format(snap_path, request.json["addon"]).split())
+    return output
+
+
+@app.route('/{}/start'.format(CLUSTER_API), methods=['POST'])
```

Comment:
Given you will need to start MicroK8s from the command line, I do not think we should allow the stopping of services via the REST api. It would be possible users may be surprised by this.

Reply:
Also, I understand you know why this call is not going to be called.

Reply:
Will remove it